Utils/azuretools-core/src/com/microsoft/azuretools/telemetrywrapper/DefaultProducer.java

```diff
@@ -0,0 +1,101 @@
+package com.microsoft.azuretools.telemetrywrapper;
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public class DefaultProducer implements Producer {
+    private static ThreadLocal<String> operIDS = new ThreadLocal<>();
+    private static final String OPERATION_NAME = "operationName";
+    private static final String OPERATION_ID = "operationId";
+    private static final String ERROR_CODE = "errorCode";
+    private static final String ERROR_MSG = "message";
+    private static final String ERROR_TYPE = "errorType";
+    private static final String DURATION = "duration";
```

Comment:
String type duration ?

Reply:
this is the duration name, not the value, the duration value is double.
map.put("duration", doubleval)

Reply:
my bad.